src/System.Private.Uri/src/System/Uri.cs

```diff
@@ -2193,13 +2191,19 @@ private unsafe ParsingError PrivateParseMinimal()
                     if (err != ParsingError.None)
                         return err;
 
+                    char hostTerminator = idx < (ushort)length ? pUriString[idx] : default(char);
+
                     // This will disallow '\' as the host terminator for any scheme that is not implicitFile or cannot have a Dos Path
-                    if ((idx < (ushort)length && pUriString[idx] == '\\') && NotAny(Flags.ImplicitFile) &&
-                        (_syntax.NotAny(UriSyntaxFlags.AllowDOSPath) ||
-                        (InFact(Flags.UncPath) && _syntax.NotAny(UriSyntaxFlags.ConvertPathSlashes))))
+                    if (hostTerminator == '\\' && NotAny(Flags.ImplicitFile) && _syntax.NotAny(UriSyntaxFlags.AllowDOSPath))
                     {
                         return ParsingError.BadAuthorityTerminator;
                     }
+
+                    // When the hostTerminator is '/' on Unix, use the UnixFile syntax (preserve backslashes)
+                    if (!IsWindowsSystem && hostTerminator == '/' && NotAny(Flags.ImplicitFile) && InFact(Flags.UncPath) && _syntax == UriParser.FileUri)
+                    {
+                        _syntax = UriParser.UnixFileUri;
+                    }
```

Comment:
If idx >= length, then hostTerminator above is being set to default(char), which means neither of these if blocks' conditions are going to match.  Seems like this could be restructured to avoid unnecessary comparisons, e.g.
```C#
if (idx < length)
{
    char hostTerminator = pUriString[idx];
    if (hostTerminator == '\\')
    {
        if (NotAny(Flags.ImplicitFile) && _syntax.NotAny(UriSyntaxFlags.AllowDOSPath))
        {
            return ParsingError.BadAuthorityTerminator;
        }
    }
    else if (!IsWindowsSystem && hostTerminator == '/' && NotAny(Flags.ImplicitFile) && InFact(Flags.UncPath) && _syntax == UriParser.FileUri)
    {
        _syntax = UriParser.UnixFileUri;
    }
}
```